eskip/eskip.go

```diff
@@ -44,13 +53,28 @@ type parsedRoute struct {
 	backend  string
 }
 
+// A CustomPredicate object represents a parsed, in-memory, route matching predicate
+// that is defined by extensions.
+type CustomPredicate struct {
```

Comment:
makes sense, in general. i see two things with that atm:
- it would probably make it a quite big pr. Rather a refactor than a feature.
- the path predicate has a distinguished role